forc-pkg/src/pkg.rs

```diff
@@ -2363,12 +2461,14 @@ pub fn check(
     sway_core::clear_lazy_statics();
     let mut namespace_map = Default::default();
     let mut source_map = SourceMap::new();
+    let compiled_map = HashMap::new();
```

Comment:
For readers of this code, seeing a collection that is forever empty might be confusing, so a comment might be useful:
```suggestion
    // During `check`, we don't compile so this stays empty.
    // In turn, default contract IDs are used when constructing the dep namespace.
    let compiled_contract_deps = HashMap::new();
```